src/main/java/mate/academy/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package mate.academy.service;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.CartItemQuantityRequestDto;
+import mate.academy.dto.CartItemRequestDto;
+import mate.academy.dto.ShoppingCartDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.CartItemMapper;
+import mate.academy.mapper.ShoppingCartMapper;
+import mate.academy.model.Book;
+import mate.academy.model.CartItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.CartItemRepository;
+import mate.academy.repository.ShoppingCartRepository;
+import mate.academy.repository.UserRepository;
+import mate.academy.repository.book.BookRepository;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public void addItemToCart(Authentication authentication,
+                              CartItemRequestDto cartItemRequestDto) {
+        User user = getUser(authentication);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseGet(() -> registerNewShoppingCart(user));
+
+        Optional<CartItem> existingCartItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId().equals(cartItemRequestDto.getBookId()))
+                .findFirst();
+        CartItem cartItemToUpdate;
+        if (existingCartItem.isPresent()) {
+            cartItemToUpdate = existingCartItem.get();
+            cartItemToUpdate.setQuantity(cartItemRequestDto.getQuantity()
+                    + cartItemToUpdate.getQuantity());
+        } else {
+            cartItemToUpdate = createNewCartItem(cartItemRequestDto, shoppingCart);
+        }
+        cartItemRepository.save(cartItemToUpdate);
+    }
```

Comment:
You make more then one request to db, make it transactional. Do the same for each method that has more then one ewqeust